packages/angular_devkit/build_angular/src/webpack/plugins/javascript-optimizer-plugin.ts

```diff
@@ -84,19 +85,37 @@ export class JavaScriptOptimizerPlugin {
           stage: compiler.webpack.Compilation.PROCESS_ASSETS_STAGE_OPTIMIZE_SIZE,
         },
         async (compilationAssets) => {
+          const cache = compilation.getCache('JavaScriptOptimizerPlugin');
```

Comment:
Would it be worthwhile to add a check here if caching is enabled and make `cache` possibly `undefined`?  With the potential advantage of skipping the cache related calls below.  I'm not sure if there is an easy and definitive way to determine cache enablement though.

Reply:
I think that should be possible via `compilation.options.experiments.cache`.

Reply:
Actually, it appears that will other experiments options can be accessed via `compilation.options.experiments`, `cache` is not.